projects/com.oracle.truffle.llvm.parser.impl/src/com/oracle/truffle/llvm/parser/impl/LLVMVisitor.java

```diff
@@ -875,21 +875,28 @@ private LLVMExpressionNode getElementPtr(LLVMExpressionNode baseAddress, Type ba
                 LLVMExpressionNode valueRef = visitValueRef(currentRef, type);
                 currentAddress = factoryFacade.createGetElementPtr(getLLVMType(type), currentAddress, valueRef, indexedTypeLength);
             } else {
-                int indexedTypeLength = LLVMTypeHelper.goIntoTypeGetLengthByte(currentType, constantIndex);
+                LLVMBaseType llvmType = LLVMTypeHelper.getLLVMType(currentType);
+                int stride;
+                if (llvmType == LLVMBaseType.ARRAY || llvmType == LLVMBaseType.ADDRESS) {
+                    stride = LLVMTypeHelper.getByteSize(currentType.getContainedType(0));
```

Comment:
I think that the following works:
stride = LLVMTypeHelper.goIntoTypeGetLengthByte(currentType, 1);